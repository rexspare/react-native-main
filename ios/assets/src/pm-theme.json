{
  "color-primary-light": "#A2FEE0",
  "color-primary-100": "#D3FAE8",
  "color-primary-200": "#AAF6D9",
  "color-primary-300": "#7AE4C5",
  "color-primary-400": "#55CAB2",
  "color-primary-500": "#36796F",
  "color-primary-600": "#1B8F8D",
  "color-primary-700": "#137078",
  "color-primary-800": "#0C5260",
  "color-primary-900": "#073D50",
  "color-primary-transparent-100": "rgba(38, 167, 153, 0.08)",
  "color-primary-transparent-200": "rgba(38, 167, 153, 0.16)",
  "color-primary-transparent-300": "rgba(38, 167, 153, 0.24)",
  "color-primary-transparent-400": "rgba(38, 167, 153, 0.32)",
  "color-primary-transparent-500": "rgba(38, 167, 153, 0.4)",
  "color-primary-transparent-600": "rgba(38, 167, 153, 0.48)",
  "color-primary-default": "$color-primary-500",
  "color-primary2-100": "#FFF1D8",
  "color-primary2-200": "#FFDEB2",
  "color-primary2-300": "#FFC88C",
  "color-primary2-400": "#FFB16F",
  "color-primary2-500": "#FF8D40",
  "color-primary2-600": "#DB6A2E",
  "color-primary2-700": "#B74B20",
  "color-primary2-800": "#933114",
  "color-primary2-900": "#7A1E0C",
  "color-primary2-transparent-100": "rgba(255, 141, 64, 0.08)",
  "color-primary2-transparent-200": "rgba(255, 141, 64, 0.16)",
  "color-primary2-transparent-300": "rgba(255, 141, 64, 0.24)",
  "color-primary2-transparent-400": "rgba(255, 141, 64, 0.32)",
  "color-primary2-transparent-500": "rgba(255, 141, 64, 0.4)",
  "color-primary2-transparent-600": "rgba(255, 141, 64, 0.48)",
  "color-primary2-default": "$color-primary2-500",
  "color-primary3-100": "#D7FDD0",
  "color-primary3-200": "#A8FBA2",
  "color-primary3-300": "#72F475",
  "color-primary3-400": "#4EEA61",
  "color-primary3-500": "#19DD43",
  "color-primary3-600": "#12BE47",
  "color-primary3-700": "#0C9F47",
  "color-primary3-800": "#078044",
  "color-primary3-900": "#046A41",
  "color-primary3-transparent-100": "rgba(25, 221, 67, 0.08)",
  "color-primary3-transparent-200": "rgba(25, 221, 67, 0.16)",
  "color-primary3-transparent-300": "rgba(25, 221, 67, 0.24)",
  "color-primary3-transparent-400": "rgba(25, 221, 67, 0.32)",
  "color-primary3-transparent-500": "rgba(25, 221, 67, 0.4)",
  "color-primary3-transparent-600": "rgba(25, 221, 67, 0.48)",
  "color-primary3-default": "$color-primary3-500",
  "color-secondary-default": "#A7A9AC",
  "color-secondary2-default": "#335150",
  "color-accent-default": "#A8B5B4",
  "color-accent2-default": "#61DB72",
  "color-accent3-default": "#22B77F",
  "color-accent4-default": "#52BC8D",
  "color-gradient1-start": "$color-primary3-default",
  "color-gradient1-end": "$color-primary-default",
  "color-gradient2-start": "$color-primary-default",
  "color-gradient2-end": "$color-primary2-default",
  "color-primary-gradient-1": "$color-primary-default",
  "color-primary-gradient-2": "#23B484",
  "color-success-100": "#EEFACD",
  "color-success-200": "#D9F59E",
  "color-success-300": "#B5E169",
  "color-success-400": "#8EC341",
  "color-success-500": "#5D9B12",
  "color-success-600": "#49850D",
  "color-success-700": "#376F09",
  "color-success-800": "#285905",
  "color-success-900": "#1D4A03",
  "color-success-transparent-100": "rgba(93, 155, 18, 0.08)",
  "color-success-transparent-200": "rgba(93, 155, 18, 0.16)",
  "color-success-transparent-300": "rgba(93, 155, 18, 0.24)",
  "color-success-transparent-400": "rgba(93, 155, 18, 0.32)",
  "color-success-transparent-500": "rgba(93, 155, 18, 0.4)",
  "color-success-transparent-600": "rgba(93, 155, 18, 0.48)",
  "color-info-100": "#CCF8FD",
  "color-info-200": "#99ECFC",
  "color-info-300": "#66D6F6",
  "color-info-400": "#40BBED",
  "color-info-500": "#0695E2",
  "color-info-600": "#0473C2",
  "color-info-700": "#0356A2",
  "color-info-800": "#013D83",
  "color-info-900": "#012B6C",
  "color-info-transparent-100": "rgba(6, 149, 226, 0.08)",
  "color-info-transparent-200": "rgba(6, 149, 226, 0.16)",
  "color-info-transparent-300": "rgba(6, 149, 226, 0.24)",
  "color-info-transparent-400": "rgba(6, 149, 226, 0.32)",
  "color-info-transparent-500": "rgba(6, 149, 226, 0.4)",
  "color-info-transparent-600": "rgba(6, 149, 226, 0.48)",
  "color-warning-100": "#FFF3CC",
  "color-warning-200": "#FFE399",
  "color-warning-300": "#FFD066",
  "color-warning-400": "#FFBC3F",
  "color-warning-500": "#FF9D00",
  "color-warning-600": "#DB7E00",
  "color-warning-700": "#B76200",
  "color-warning-800": "#934900",
  "color-warning-900": "#7A3800",
  "color-warning-transparent-100": "rgba(255, 157, 0, 0.08)",
  "color-warning-transparent-200": "rgba(255, 157, 0, 0.16)",
  "color-warning-transparent-300": "rgba(255, 157, 0, 0.24)",
  "color-warning-transparent-400": "rgba(255, 157, 0, 0.32)",
  "color-warning-transparent-500": "rgba(255, 157, 0, 0.4)",
  "color-warning-transparent-600": "rgba(255, 157, 0, 0.48)",
  "color-danger-100": "#FDE3D1",
  "color-danger-200": "#FBC0A5",
  "color-danger-300": "#F39376",
  "color-danger-400": "#E76953",
  "color-danger-500": "#D82C20",
  "color-danger-600": "#B9171A",
  "color-danger-700": "#9B101E",
  "color-danger-800": "#7D0A1F",
  "color-danger-900": "#670620",
  "color-danger-transparent-100": "rgba(216, 44, 32, 0.08)",
  "color-danger-transparent-200": "rgba(216, 44, 32, 0.16)",
  "color-danger-transparent-300": "rgba(216, 44, 32, 0.24)",
  "color-danger-transparent-400": "rgba(216, 44, 32, 0.32)",
  "color-danger-transparent-500": "rgba(216, 44, 32, 0.4)",
  "color-danger-transparent-600": "rgba(216, 44, 32, 0.48)",
  "color-info-gradient-1": "#57ABF6",
  "color-info-gradient-2": "#11CDD5",
  "color-danger-gradient-1": "#E46C81",
  "color-danger-gradient-2": "#FC5457",
  "color-basic-control-transparent-700": "rgba(255, 255, 255, 0.56)",
  "color-basic-control-transparent-800": "rgba(255, 255, 255, 0.64)",
  "color-basic-control-transparent-900": "rgba(255, 255, 255, 0.72)",
  "color-basic-control-transparent-1000": "rgba(255, 255, 255, 0.80)",
  "color-basic-control-transparent-1100": "rgba(255, 255, 255, 0.88)",
  "color-grey-gradient-1": "#465A64",
  "color-grey-gradient-2": "#B0BEC4",
  "grey-0": "#f2f5f8",
  "grey-100": "#CFD8DC",
  "grey-200": "#B0BEC4",
  "grey-300": "#91A4AD",
  "grey-400": "#79909B",
  "grey-600": "#556E79",
  "grey-700": "#465A64",
  "color-grey-0": "#f2f5f8",
  "color-grey-100": "#CFD8DC",
  "color-grey-200": "#B0BEC4",
  "color-grey-300": "#91A4AD",
  "color-grey-400": "#79909B",
  "color-grey-500": "#79909B",
  "color-grey-600": "#556E79",
  "color-grey-700": "#465A64",
  "new-black": "#263238",
  "grey-0-transparent": "rgba(242,245,248, 0.56)",
  "grey-100-transparent": "rgba(207,216,220, 0.56)",
  "grey-200-transparent": "rgba(176,190,196, 0.56)",
  "grey-300-transparent": "rgba(145,164,173, 0.56)",
  "grey-400-transparent": "rgba(121,144,155, 0.56)",
  "grey-700-transparent": "rgba(70,90,100, 0.56)",
  "new-black-transparent": "rgba(38,50,56, 0.56)",
  "unit-vacant": "#FD8B83",
  "unit-listed": "#48A7F2",
  "comment": "text-basic-color: #465A64",
  "color-basic-disabled": "$color-basic-control-transparent-700",
  "color-primary-disabled": "$color-primary-transparent-700",
  "comment2": "text-disabled-color: $color-basic-control-transparent-900",
  "text-hint-color": "$grey-400"
}
