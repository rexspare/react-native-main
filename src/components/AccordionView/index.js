import Text from 'components/Text';
import React, { useState } from 'react';
import { View } from 'react-native';
import Accordion from 'react-native-collapsible/Accordion';
import Box from 'components/Box';
import { Icon } from '@ui-kitten/components';
import { styles } from './styles';

const AccordionView = ({ sections }) => {

  const [state, setState] = useState({
    activeSections: [],
  })

  const _renderHeader = (section, i, isActive) => {
    return (
      <View style={{
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        width: '100%',
        ...styles.header,
        ...(i === sections.length - 1 && styles.headerLast)
      }}
      >
        <Text style={styles.headerText}>{section.title}</Text>
        <Icon
          name={isActive ? 'minus' : 'plus'}
          width={15}
          height={15}
        />
      </View>
    );
  };

  const _renderContent = (section, i) => {
    return (
      <View style={styles.content}>
        <Box>{section.content}</Box>
      </View>
    );
  };

  const _updateSections = (activeSections) => {
    setState({ activeSections });
  };
  return (
    <Accordion
      sections={sections}
      activeSections={state.activeSections}
      renderHeader={_renderHeader}
      renderContent={_renderContent}
      onChange={_updateSections}
      underlayColor="#eee"
    />
  )
};


export default AccordionView;