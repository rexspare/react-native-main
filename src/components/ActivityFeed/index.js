import React, { useRef, useState } from 'react';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { useNavigation } from '@react-navigation/native';
import activityFeedQuery from 'queries/activity/ActivityFeed.gql';
import { useGetListProps } from 'hooks/useGetListProps';
import { useIsOpen } from 'hooks/useIsOpen';
import { useKeyboardEffect } from 'hooks/useKeyboardEffect';
import { renderSectionHeader as _renderSectionHeader } from 'pages/tasks/TaskList';
import Box from 'components/Box';
import InfiniteFlatList from 'components/InfiniteFlatList';
import Text from 'components/Text';
import ActivityFeedItem from './ActivityFeedItem';
import CommentBox from './CommentBox';
import { t } from 'helpers/react';
import {
  sectionExtractor,
  _getSections,
  renderSectionHeader,
} from '../../helpers/list';
import { styles } from './styles';

const ActivityFeed = ({
  feedId,
  listProps: _listProps,
  onLinkPress,
  keyboardPadding = true,
  comments = true,
  limit = 18,
  contentContainerStyle = { justifyContent: 'space-between', height: '100%' },
}) => {
  const [scrollRef, setScrollRef] = useState(null);
  const [sections, setSections] = useState([]);
  const nav = useNavigation();
  const listRef = useRef();

  const {
    isOpen: keyboardIsOpen,
    open: setKeyboardIsOpen,
    close: setKeyboardIsClosed,
  } = useIsOpen();

  useKeyboardEffect({
    onIsOpen: setKeyboardIsOpen,
    onIsClosed: setKeyboardIsClosed,
  });

  const listProps = useGetListProps(
    {
      dataKey: 'activities',
      renderItem: ({ item }) => (
        <ActivityFeedItem item={item} nav={nav} onLinkPress={onLinkPress} />
      ),
      variables: { feedId },
      limit: limit,
      onResCallback: res => getSections(res?.data),
      sections,
      sectionExtractor: item =>
        sectionExtractor({ date: item?.createdAt, df: 'MMMM dd yyyy' }),
      renderSectionHeader: section =>
        renderSectionHeader(section, styles.sectionHeaderText),
      ..._listProps,
    },
    [sections, sections.length, _listProps],
  );

  const getSections = _data => {
    const sections = _getSections(
      _data?.activities?.edges,
      'createdAt',
      'MMMM dd yyyy',
    );
    return setSections(sections);
  };

  return (
    <>
      <KeyboardAwareScrollView
        innerRef={ref => setScrollRef(ref)}
        behavior="position"
        onContentSizeChange={() =>
          keyboardIsOpen && scrollRef.scrollToEnd({ animated: true })
        }
        contentContainerStyle={contentContainerStyle}>
        <Box flex={1}>
          <InfiniteFlatList
            query={activityFeedQuery}
            ref={listRef}
            {...listProps}
            ListEmptyComponent={
              <Text category="h6" textAlign="center" appearance="hint">
                NO ACTIVITY
              </Text>
            }
          />
        </Box>
        {t(
          comments,
          <Box
            height={60}
            marginBottom={keyboardPadding && keyboardIsOpen ? 36 : 0}
            alignSelf={'flex-end'}>
            <CommentBox
              feedId={feedId}
              refreshFeed={() => listRef?.current?.refresh()}
            />
          </Box>,
        )}
      </KeyboardAwareScrollView>
    </>
  );
};

export default ActivityFeed;
