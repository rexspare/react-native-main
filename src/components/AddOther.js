import React, { useState } from 'react';
import { Button } from '@ui-kitten/components';
import BottomHalfModal from 'components/BottomHalfModal';
import Icon from './Icon';
import Input from './Input';
import { colors } from 'styles/theme';

const AddOtherModal = ({ children, onPress, ...props }) => {
  const [inputValue, setInputValue] = useState('');

  return (
    <BottomHalfModal
      displayDone={false}
      styles={{
        headerTxt: {
          textTransform: 'uppercase',
          fontWeight: 'bold',
        },
      }}
      elStyle
      {...props}>
      <Input
        labelStyle={{ color: colors['gray scale/40'], fontWeight: '500' }}
        label="Other"
        isUppercase
        onChangeText={val => setInputValue(val)}
        minHeight={48}
        icon={Icon('expandInput', 'pm')}
      />
      <Button
        style={{ borderRadius: 12, marginTop: 40 }}
        textStyle={{ fontSize: 16, textTransform: 'uppercase' }}
        onPress={() => !!inputValue && onPress(inputValue)}>
        Save
      </Button>
    </BottomHalfModal>
  );
};

export default AddOtherModal;
