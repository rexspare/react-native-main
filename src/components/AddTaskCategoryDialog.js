import React from 'react';
import Dialog from './Dialog';
import Input from './Input';

import createTaskCategoryMutation from 'queries/tasks/createTaskCategory.gql';
import {useMutation} from 'urql';

const AddTaskCategoryDialog = ({visible, onHide, onAdd}) => {
  const [name, setName] = React.useState(null);
  const [error, setError] = React.useState(null);
  const [createRes, executeMutation] = useMutation(createTaskCategoryMutation);

  React.useEffect(() => {
    setName(null);
    setError(null);
  }, [visible]);

  const onSubmit = React.useCallback(() => {
    const create = async () => {
      const res = await executeMutation({name});
      if (res?.data?.createTaskCategory?.category) {
        onAdd({
          id: res?.data?.createTaskCategory?.category.id,
          name,
        });
        onHide();
      } else if (res.error) {
        setError(res.error);
      }
    };
    create();
  }, [executeMutation, name, onAdd, onHide]);

  return (
    <Dialog
      visible={!!visible}
      onHide={onHide}
      title="New Task Folder"
      content={
        error ? (
          'Failed to create task folder.'
        ) : (
          <Input
            minWidth={180}
            placeholder="Name"
            value={name}
            onChangeText={setName}
            autoFocus
          />
        )
      }
      buttons={[
        {
          children: 'Create',
          gradient: true,
          shape: 'circle',
          hide: false,
          disabled: !name?.length || createRes.fetching,
          onPress: onSubmit,
        },
      ]}
    />
  );
};

export default AddTaskCategoryDialog;
