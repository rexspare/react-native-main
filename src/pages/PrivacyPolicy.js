import React from 'react';
import SafeAreaView from 'components/SafeAreaView';
import {Layout} from '@ui-kitten/components';
import Box from 'components/Box';
import Header from 'components/Header';
import Text from 'components/Text';
import {ScrollView} from 'react-native-gesture-handler';

export default function PrivacyPolicy({navigation}) {
  return (
    <Box flex={1} as={Layout} pb={20}>
      <Box flex={1} as={SafeAreaView} forceInset={{top: 'always'}}>
        <Header
          actions={[
            {
              icon: 'arrow-ios-back',
              left: true,
              onPress: () => navigation.goBack(),
            },
          ]}
          alignment="center"
          title="Privacy Policy"
          divider
        />
        <Box as={ScrollView} alwaysBounceVertical={false}>
          <Text category="p2" py="3" px="4">
            Sed ut perspiciatis unde omnis iste natus error sit voluptatem
            accusantium doloremque laudantium, totam rem aperiam, eaque ipsa
            quae ab illo inventore veritatis et quasi architecto beatae vitae
            dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit
            aspernatur aut odit aut fugit.{'\n\n'}Sed quia consequuntur magni
            dolores eos qui ratione voluptatem sequi nesciunt. Neque porro
            quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur,
            adipisci velit, sed quia non numquam eius modi tempora incidunt ut
            labore et dolore magnam aliquam quaerat voluptatem.{'\n\n'}Neque
            porro quisquam est, qui dolorem ipsum quia dolor sit amet,
            consectetur, adipisci velit, sed quia non numquam eius modi tempora
            incidunt ut labore et dolore magnam aliquam quaerat voluptatem.
          </Text>
        </Box>
      </Box>
    </Box>
  );
}
