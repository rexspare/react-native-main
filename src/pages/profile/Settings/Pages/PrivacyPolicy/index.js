import React from 'react';
import { Dimensions } from 'react-native';
import { WebView } from 'react-native-webview';
import Config from 'react-native-config';

const PrivacyPolicy = () => {
  const injectedJavaScript = `
  const body = document.getElementsByTagName('body');
    for (var i = 0; i < body.length; i++) {
        body[i].style.fontFamily  = 'Roboto';
    }

  const paragraphs = document.getElementsByTagName('p');
    for (var i = 0; i < paragraphs.length; i++) {
      paragraphs[i].style.fontFamily  = 'Roboto';
    }

  const span = document.getElementsByTagName('span');
    for (var i = 0; i < span.length; i++) {
      span[i].style.fontFamily  = 'Roboto';
    }

  const p = document.getElementsByTagName('p');
    for (var i = 0; i < p.length; i++) {
      p[i].style.fontFamily  = 'Roboto';
      p[i].style.lineHeight = 2
    }
  `;

  return (
    <WebView
      nestedScrollEnabled
      startInLoadingState={true}
      style={{ marginTop: -80, height: Dimensions.get('window').height }}
      source={{
        uri: `${Config.WEBSITE_URL}privacyPolicy?mode=webview_content_only`,
      }}
      injectedJavaScript={injectedJavaScript}
    />
  );
};

export default PrivacyPolicy;
