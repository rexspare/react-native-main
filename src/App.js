import React from 'react';
import { ApplicationProvider, IconRegistry } from '@ui-kitten/components';
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import { mapping, light as baseTheme } from '@eva-design/eva';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import Toast from 'react-native-toast-message';
import { toastConfig } from 'config/toast-config';

import AsyncStorage from '@react-native-community/async-storage';

import RootNavigator from 'navigations';
import StorageProvider from 'providers/storage';
import ThemeProvider from 'providers/theme';
import AuthProvider from 'providers/auth';

import RefreshStateProvider from 'providers/refresh';

import NotificationsProvider from 'providers/notifications';

import { default as appTheme } from './pm-theme.json';
import { default as appMapping } from './pm-mapping';
import PmIconsPack from 'components/PmIconPackage';
import { StatusBar } from 'react-native';
import FABProvider from 'providers/fab.js';
import GraphQLProvider from 'providers/graphql.js';
import FilterContext from 'providers/filter';

console.disableYellowBox = true;

delete mapping.components.Input.appearances.default.variantGroups.size.small
  .borderRadius;
delete mapping.components.Input.appearances.default.variantGroups.size.medium
  .borderRadius;
delete mapping.components.Input.appearances.default.variantGroups.size.large
  .borderRadius;

const theme = { ...baseTheme, ...appTheme };

if (!__DEV__) {
  const noop = () => { };
  console.log = noop;
  console.warn = noop;
  console.error = noop;
}
else {
  import('../ReactotronConfig').then(() => console.log('Reactotron Configured'))

}
console.log(require('react-native-config').default);

const App = () => (
  <>
    <IconRegistry icons={[EvaIconsPack, PmIconsPack]} />
    <StatusBar
      translucent
      backgroundColor="transparent"
      barStyle="light-content"
    />
    <GraphQLProvider>
      <ApplicationProvider
        mapping={mapping}
        customMapping={appMapping}
        theme={theme}>
        <SafeAreaProvider>
          <StorageProvider.Provider value={AsyncStorage}>
            <AuthProvider.Provider>
                <RefreshStateProvider.Provider>
                  <NotificationsProvider.Provider>
                    <ThemeProvider>
                      <FilterContext.Provider>
                        <FABProvider>
                          <RootNavigator />
                          <Toast config={toastConfig} />
                        </FABProvider>
                      </FilterContext.Provider>
                    </ThemeProvider>
                  </NotificationsProvider.Provider>
                </RefreshStateProvider.Provider>
            </AuthProvider.Provider>
          </StorageProvider.Provider>
        </SafeAreaProvider>
      </ApplicationProvider>
    </GraphQLProvider>
  </>
);

export default App;
