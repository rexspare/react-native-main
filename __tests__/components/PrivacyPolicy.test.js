import React from 'react';
import { render } from '@testing-library/react-native';
import PrivacyPolicy from '../../src/pages/PrivacyPolicy';
import { ApplicationProvider } from '@ui-kitten/components';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { mapping, light as baseTheme } from '@eva-design/eva';
import { default as appMapping } from '../../src/pm-mapping';
import { default as appTheme } from '../../src/pm-theme.json';

jest.mock('react-native-gesture-handler', () => {
    const ScrollView = require('react-native').ScrollView;
    return { ScrollView };
});



describe('Privacy Policy', () => {
    const navigation = { dispatch: jest.fn(), navigate: jest.fn() };
    const theme = { ...baseTheme, ...appTheme };
    test('renders correctly', () => {
        const output = render(
            <ApplicationProvider
                mapping={mapping}
                customMapping={appMapping}
                theme={theme}>
                <SafeAreaProvider>
                    <PrivacyPolicy navigation={navigation} />
                </SafeAreaProvider>
            </ApplicationProvider>
        );
        expect(output).toMatchSnapshot();
    });
});