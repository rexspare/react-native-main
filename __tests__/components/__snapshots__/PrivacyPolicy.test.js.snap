// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Privacy Policy renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
